scc/light_client/server.go

```diff
@@ -173,3 +177,44 @@ func (s server) getBlockCertificates(first idx.Block, maxResults uint64) ([]cert
 	}
 	return certs, nil
 }
+
+// GetAccountProof returns the account proof corresponding to the
+// given address at the given height.
+//
+// Parameters:
+// - address: The address of the account.
+// - height: The block height of the state.
+//
+// Returns:
+// - AccountInfo: The AccountInfo of the account at the given height.
+// - error: Not nil if the provider failed to obtain the requested account proof.
+func (s server) getAccountProof(address common.Address, height idx.Block) (carmen.WitnessProof, error) {
+	heightString := fmt.Sprintf("0x%x", height)
+	if height == LatestBlock {
+		heightString = "latest"
+	}
+	var result struct {
+		AccountProof []string
+	}
+	err := s.client.Call(
+		&result,
+		"eth_getProof",
+		fmt.Sprintf("%v", address),
+		[]string{fmt.Sprintf("%v", common.Hash{})},
+		heightString,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// decode elements for the proof.
+	elements := []carmen.Bytes{}
+	for _, element := range result.AccountProof {
+		data, err := hexutil.Decode(element)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode proof element: %v", err)
+		}
+		elements = append(elements, immutable.NewBytes(data))
+	}
+	return carmen.CreateWitnessProofFromNodes(elements...), nil
```

Comment:
I think this is a good place to check the resulting proof for self-consistency using its [IsValid](https://github.com/0xsoniclabs/carmen/blob/712d1a76e9297cab04620398f44258d5a2b0cd8e/go/carmen/proof.go#L48) function. If the proof is invalid, an error should be returned instead. Wdyt?